guidelines/faster-review.md

```diff
@@ -0,0 +1,115 @@
+# Faster Code Review Guideline
+
+Kyma is quite a large project with many areas, each has a group of core developers
+who are working hard on developing that area of Kyma. While that means they are
+working hard on new features and enhancements, they also work on doing code
+reviews for PRs coming from different contributors from the community and it is
+crucial that the code review process of new PRs becomes quick and efficient for
+both parties. This guide defines a set of principles and practices that can help
+achieving such a goal.
+
+## 0. Familiarize yourself with project conventions
+
+- [CONTRIBUTING.md](../CONTRIBUTING.md)
+- [Naming conventions](naming.md)
+- [Content strategy](content-guidelines/content-strategy.md)
+- [Content guidelines](content-guidelines)
+- [Coding standards](coding-standards)
+
+## 1. Before coding communication
+
+We recommend these steps before you jump into coding that nice feature you want
+
+1. Make sure you have a GitHub issue created for the work you want to do that
+   - Follows one of the issues templates and make sure to describe well the
+    the changes you intend to introduce.
+   - If possible identify in the issue the relevant area label and components
+    you might touch and change.
+2. Communicate that issue upfront on the slack channel of the relevant area and
+start a discussion there to have it pre-reviewed.
+3. Ask the area group to assign you an issue buddy from the CODEOWNERS.
+    >Since reviewing a PR always needs context, it is best if the whole pre-review,
+assistance, and code review stays with a single area member who is also a
+CODEOWNER. We call that person an `issue buddy`.
+
+## 2. Don't build a cathedral in one PR
+
+Are you sure you will have the CODEOWNERS buy-in and support in a single PR?
+Are you sure they won't have doubts about your technique and whether this is the
+best way to tackle that issue? Are you willing to bet a few days or weeks of
+work on it?  If in doubt, consider one of the following:
+
+1. Start with a design proposal and have it reviewed.
+2. Create WIP PR where you get early feedback on initial implementation.
```

Comment:
Since a couple of weeks Github offers the "draft pr" feature. You might want to mention that together with or instead WIP label.